curator-x-discovery/src/test/java/org/apache/curator/x/discovery/details/DiscoveryPathConstructorImplTest.java

```diff
@@ -0,0 +1,59 @@
+package org.apache.curator.x.discovery.details;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class DiscoveryPathConstructorImplTest
+{
+
+    @Test
+    public void testCanonicalDiscoveryPathConstructor()
+    {
+        Assert.expectThrows(IllegalArgumentException.class, new Assert.ThrowingRunnable()
+            {
+                @Override
+                public void run() throws Throwable {
+                    new DiscoveryPathConstructorImpl(null);
+                }
+            }
+        );
+    }
+
+    @Test
+    public void testGetBasePath() throws Exception
+    {
+        Assert.assertEquals(new DiscoveryPathConstructorImpl("/foo/bar").getBasePath(), "/foo/bar");
+        Assert.assertEquals(new DiscoveryPathConstructorImpl("foo/bar").getBasePath(), "/foo/bar");
```

Comment:
I worried if we should forbid this pattern, said the base dir passed without a leading slash, instead of tolerate it.

It seems somehow informal especially for `foo/bar/` patterns.